### PR TITLE
Add Checkstyle to POM with Google settings advice

### PR DESCRIPTION
**Pull request by nickcmaynard**
### PR checklist
- [x] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.
### Description of the PR

Add Checkstyle to the Maven build, reporting noncompliance, but not enforcing style.

### CODE REVIEW

**Comment by cbornet** [comment]
The code is currently indented with 4 spaces so I don't think it follows google's style.

**Comment by nickcmaynard** [comment]
To be clear, the config has been adjusted to account for 4-space indentation.

**Comment by ePaul** [comment]
> To be clear, the config has been adjusted to account for 4-space indentation.

Maybe also state that in the comment at the begin of the file, so people don't think this is the original one?

**Comment by nickcmaynard** [comment]
Good idea - done.

File: google_checkstyle.xml [hunk 1]
```diff
@@ -1,6 +1,11 @@
 <?xml version="1.0"?>
 <!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
 
+<!-- 
+    Adjusted from original Google Checkstyle config to match requirements for swagger-codegen
+    See detailed comments next to changes
```
Comment (nickcmaynard) [inline google_checkstyle.xml]: I shall add a link to the original file.  However, I shan't be paying a great deal of attention to comments on grammar in a *comment* on a *configuration file* where it's patently obvious what's going on.

**Comment by nickcmaynard** [comment]
@wing328 Any thoughts on merging this?

**Comment by ePaul** [comment]
Looks good to me. :+1:

**Comment by wing328** [comment]
@nickcmaynard I did some tests and found the warnings (which are expected).

Thanks for the contribution.

**Pull request merged**